File: vcc-sdk-java/src/main/java/com/ctrip/ccard/creditcard/vcc/bean/V2/FxCreateRequest.java
package com.ctrip.ccard.creditcard.vcc.bean.V2;

import java.math.BigDecimal;

public class FxCreateRequest {
    /**
     * API版本
     */
    private String serviceVersion;
    /**
     * 请求流水号
     */
    private String requestId;
    /**
     * 客户ID
     */
    private String customerId;
    /**
     * 卖出币种
     */
    private String sellCurrency;
    /**
     * 买入币种
     */
    private String buyCurrency;
    /**
     * 换汇方向
     * 0-指定卖出
     * 1-指定买入
     */
    private Integer fxDirection;
    /**
     * 交易金额
     */
    private BigDecimal fxAmount;
    /**
     * 汇率ID
     */
    private String quoteId;


    public String getServiceVersion() {
        return serviceVersion;
    }

    public void setServiceVersion(String serviceVersion) {
        this.serviceVersion = serviceVersion;
    }

    public String getRequestId() {
        return requestId;
    }

    public void setRequestId(String requestId) {
        this.requestId = requestId;
    }

    public String getCustomerId() {
        return customerId;
    }

    public void setCustomerId(String customerId) {
        this.customerId = customerId;
    }

    public String getSellCurrency() {
        return sellCurrency;
    }

    public void setSellCurrency(String sellCurrency) {
        this.sellCurrency = sellCurrency;
    }

    public String getBuyCurrency() {
        return buyCurrency;
    }

    public void setBuyCurrency(String buyCurrency) {
        this.buyCurrency = buyCurrency;
    }

    public Integer getFxDirection() {
        return fxDirection;
    }

    public void setFxDirection(Integer fxDirection) {
        this.fxDirection = fxDirection;
    }

    public BigDecimal getFxAmount() {
        return fxAmount;
    }

    public void setFxAmount(BigDecimal fxAmount) {
        this.fxAmount = fxAmount;
    }

    public String getQuoteId() {
        return quoteId;
    }

    public void setQuoteId(String quoteId) {
        this.quoteId = quoteId;
    }

    @Override
    public String toString() {
        return "FxCreateRequest{" +
                "serviceVersion='" + serviceVersion + '\'' +
                ", requestId='" + requestId + '\'' +
                ", customerId='" + customerId + '\'' +
                ", sellCurrency='" + sellCurrency + '\'' +
                ", buyCurrency='" + buyCurrency + '\'' +
                ", fxDirection=" + fxDirection +
                ", fxAmount=" + fxAmount +
                ", quoteId='" + quoteId + '\'' +
                '}';
    }
}

File: php-sdk/src/v2/model/core/FxCreateRequest.php
<?php declare(strict_types=1);

namespace v2\model\core;

use JsonSerializable;
use v2\model\BaseRequest;

class FxCreateRequest extends BaseRequest implements JsonSerializable {
    /**
     * 卖出币种
     * @var string sell currency
     */
    private $sellCurrency;
    /**
     * 买入币种
     * @var string buy currency
     */
    private $buyCurrency;
    /**
     * 交易方向
     * @var int trading direction
     */
    private $fxDirection;
    /**
     * 交易金额
     * @var float trading amount
     */
    private $fxAmount;

      /**
         * 汇率id
         * @var string|null
    */
    private $quoteId;

    public function getSellCurrency(): string {
        return $this->sellCurrency;
    }

    public function setSellCurrency(string $sellCurrency): void {
        $this->sellCurrency = $sellCurrency;
    }

    public function getBuyCurrency(): string {
        return $this->buyCurrency;
    }

    public function setBuyCurrency(string $buyCurrency): void {
        $this->buyCurrency = $buyCurrency;
    }

    public function getFxDirection(): int {
        return $this->fxDirection;
    }

    public function setFxDirection(int $fxDirection): void {
        $this->fxDirection = $fxDirection;
    }

    public function getFxAmount(): float {
        return $this->fxAmount;
    }

    public function setFxAmount(float $fxAmount): void {
        $this->fxAmount = $fxAmount;
    }

    public function getQuoteId(): string {
        return $this->quoteId;
    }

    public function setQuoteId(string $quoteId): void {
         $this->quoteId = $quoteId;
    }

    public function jsonSerialize(): object {
        return (object)array_filter(get_object_vars($this), function ($value) {
            return $value !== null;
        });
    }
}